### DISCUSSION TITLE
Is there a method to click on a alert button create by ionic AlertController ?

### DISCUSSION THREAD

**Discussion by insistedMarin** [cypress-io/cypress#24942] [Component Testing]
let alert = this.alertCtrl.create({
    title: 'Confirm purchase',
    message: 'Do you want to buy this book?',
    buttons: [
      {
        text: 'Cancel',
        role: 'cancel',
        handler: () => {
          console.log('Cancel clicked');
        }
      },
      {
        text: 'Buy',
        handler: () => {
          console.log('Buy clicked');
        }
      }
    ]
  });
  alert.present();

I try to use .click() or .trigger('click') to click on the button namd 'Buy', but it shows 
Timed out retrying after 4050ms: cy.click() failed because this element:

\<span class="alert-button-inner sc-ion-alert-ios">Buy\</span>

has CSS pointer-events: none

pointer-events: none prevents user mouse interaction.

**Comment by insistedMarin** [comment]
it needs to use .click({force: true})

**Comment by stavares843** [comment]
does `window:prompt` works?

https://applitools.com/blog/testing-browser-alerts-confirmations-prompts-cypress/

in case that's helpful 🔨
